File: ABC/ABC235/B/csharp/Program.cs
using System;
using System.Linq;

class Program
{
    static void Main(string[] args)
    {
        var _ = Console.ReadLine();
        var h = Console.ReadLine().Split(" ").Select(int.Parse).ToArray();
        var ans = h[0];
        for (int i = 1; i < h.Length; i++)
        {
            if (ans >= h[i]) break;
            ans = h[i];
        }
        Console.WriteLine(ans);
    }
}

File: ABC/ABC235/B/fsharp/Program.fs
let n = stdin.ReadLine() |> int
let h = stdin.ReadLine().Split(" ") |> Array.map int
let ans =
    let rec f (i, acc) =
        if i < n then
            if acc < h.[i] then f (i+1, h.[i]) else acc
        else
            acc
    f (1, h.[0])
printfn "%d" ans
